From d181787a153a3645e5b7ec654c48dc9dd09c25c9 Mon Sep 17 00:00:00 2001
From: Zahari Doychev <zdoychev@maxlinear.com>
Date: Wed, 7 Oct 2020 11:29:03 +0200
Subject: [PATCH] net: flower: fl_dump add missing eth_type attr

The TCA_FLOWER_KEY_ETH_TYPE attribute key seems inot to be not set currently.
It seems this was removed by mistake from previous backports. Without this
attribute the filter dumps do not show the icmp type and code values.
---
 net/sched/cls_flower.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/net/sched/cls_flower.c b/net/sched/cls_flower.c
index ffead9469c11..79efafcf1a9f 100644
--- a/net/sched/cls_flower.c
+++ b/net/sched/cls_flower.c
@@ -995,7 +995,10 @@ static int fl_dump(struct net *net, struct tcf_proto *tp, unsigned long fh,
 			    sizeof(key->eth.dst)) ||
 	    fl_dump_key_val(skb, key->eth.src, TCA_FLOWER_KEY_ETH_SRC,
 			    mask->eth.src, TCA_FLOWER_KEY_ETH_SRC_MASK,
-			    sizeof(key->eth.src)))
+			    sizeof(key->eth.src)) ||
+	    fl_dump_key_val(skb, &key->basic.n_proto, TCA_FLOWER_KEY_ETH_TYPE,
+			    &mask->basic.n_proto, TCA_FLOWER_UNSPEC,
+			    sizeof(key->basic.n_proto)))
 		goto nla_put_failure;
 
 
